src/jit/lsraarmarch.cpp

```diff
@@ -738,8 +738,22 @@ void Lowering::TreeNodeInfoInitPutArgStk(GenTreePutArgStk* argNode, fgArgTabEntr
     else
     {
 #ifdef _TARGET_ARM_
+
+#ifndef ARM_SOFTFP
         // We must not have a multi-reg struct; double uses 2 slots and isn't a multi-reg struct
         assert((info->numSlots == 1) || ((info->numSlots == 2) && (putArgChild->TypeGet() == TYP_DOUBLE)));
+#else  // ARM_SOFTFP
+        // The `double` types have been transformed to `long` on armel.
+        if ((info->numSlots == 2) && (putArgChild->TypeGet() == TYP_LONG))
+        {
+            argNode->gtLsraInfo.srcCount = 2;
+        }
+        else
+        {
+            assert(info->numSlots == 1);
+        }
+#endif // ARM_SOFTFP
+
```

Comment:
@CarolEidt I first tried something like your code, but updating srcCount was needed for ARM_SOFTFP. So here's what I've got. Would this look better than the current one? Or would there be a better implementation?

```cpp
#ifdef _TARGET_ARM_

#ifdef ARM_SOFTFP
        // The `double` types have been transformed to `long` on armel.
        var_types doubleType = TYP_LONG;
#else
        var_types doubleType = TYP_DOUBLE:
#endif

        // We must not have a multi-reg struct; double uses 2 slots and isn't a multi-reg struct
        const bool isDouble = (info->numSlots == 2) && (putArgChild->TypeGet() == doubleType);
        assert((info->numSlots == 1) || isDouble);

#ifdef ARM_SOFTFP
        if (isDouble)
        {
            argNode->gtLsraInfo.srcCount = 2;
        }
#endif

#else  // !_TARGET_ARM_
        // We must not have a multi-reg struct
        assert(info->numSlots == 1);
#endif // !_TARGET_ARM_
```